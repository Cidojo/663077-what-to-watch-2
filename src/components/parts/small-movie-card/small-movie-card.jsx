import * as React from 'react';
import {Link} from 'react-router-dom';

import MovieCardImageWrapped from './../movie-card-image/movie-card-image.jsx';
import {movieCardPropTypes} from './../../../global-custom-types.js';

const SmallMovieCard = (props) => {
  const {
    card
  } = props;

  return (
    <article className="small-movie-card catalog__movies-card">
      <Link to={`/films/${card.id}`}>
        <MovieCardImageWrapped
          card={card}
        />
      </Link>
      <h3 className="small-movie-card__title">
        <Link
          className="small-movie-card__link"
          to={`/films/${card.id}`}
        >
          {card.name}
        </Link>
      </h3>
    </article>
  );
};

SmallMovieCard.propTypes = {
  card: movieCardPropTypes
};

SmallMovieCard.defaultProps = {
  card: {}
};

export {SmallMovieCard};
