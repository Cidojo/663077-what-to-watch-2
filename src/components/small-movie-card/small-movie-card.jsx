import * as React from 'react';
import * as PropTypes from 'prop-types';
import {MovieCardImage} from '../movie-card-image/movie-card-image.jsx';
import {VideoPlayer} from '../video-player/video-player.jsx';
import {movieCardPropTypes} from './../../global-custom-types';
import withPlayer from './../../hocs/with-player/with-player.jsx';

const MovieCardImageWithPlayer = withPlayer(MovieCardImage, VideoPlayer);

const SmallMovieCard = (props) => {
  const {
    card,
    onMovieCardTitleClick
  } = props;

  const _handleMovieCardTitleClick = () => {
    onMovieCardTitleClick(card.id);
  };

  return (
    <article className="small-movie-card catalog__movies-card">
      <MovieCardImageWithPlayer
        src={card.src}
        muted={true}
        poster={card.posterSrc}
      />
      <h3 className="small-movie-card__title">
        <a
          className="small-movie-card__link"
          href={card.link}
          onClick={_handleMovieCardTitleClick}
        >
          {card.title}
        </a>
      </h3>
    </article>
  );
};

SmallMovieCard.propTypes = {
  card: movieCardPropTypes.isRequired,
  onMovieCardTitleClick: PropTypes.func.isRequired
};

SmallMovieCard.defaultProps = {
  card: {},
  onMovieCardTitleClick: () => {},
};

export {SmallMovieCard};
