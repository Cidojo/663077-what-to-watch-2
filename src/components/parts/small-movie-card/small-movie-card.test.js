import * as React from 'react';
import {SmallMovieCard} from './small-movie-card.jsx';
import * as renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

const props = {
  card: {}
};

it(`should render SmallMovieCard component without errors`, () => {
  const tree = renderer.create(<BrowserRouter><SmallMovieCard {...props} /></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
