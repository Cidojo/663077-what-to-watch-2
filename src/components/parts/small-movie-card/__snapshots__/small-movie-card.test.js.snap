// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SmallMovieCard component without errors 1`] = `
<article
  className="small-movie-card catalog__movies-card"
>
  <div
    className="small-movie-card__image"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <video
      className="player__video"
      muted={true}
      poster=""
      src=""
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/films/undefined"
      onClick={[Function]}
    />
  </h3>
</article>
`;
